azurecompute/src/main/java/org/jclouds/azurecompute/compute/functions/DeploymentToNodeMetadata.java

```diff
@@ -117,8 +118,11 @@ public NodeMetadata apply(final Deployment from) {
       // TODO: CloudService name is required (see JCLOUDS-849): waiting for JCLOUDS-853.
       final CloudService cloudService = api.getCloudServiceApi().get(from.name());
       if (cloudService != null) {
+         final String location = cloudService.location() != null
+                 ? cloudService.location()
+                 : api.getAffinityGroupApi().get(cloudService.affinityGroup()).location();
```

Comment:
Well, to be absolutely honest, I'm not sure -- but the `CloudService` javadoc suggests that these two are mutually exclusive.